src/markdown-pages/install-with-kurl/migrating-csi.md

```diff
@@ -3,73 +3,206 @@ path: "/docs/install-with-kurl/migrating-csi"
+
+  See the following for requirements specific to Rook or OpenEBS: 
+  - **Rook Ceph**: See [Rook Add-on](/add-ons/rook) in the kURL documentation and [Hardware Recommendations](https://docs.ceph.com/en/quincy/start/hardware-recommendations/) in the Ceph documentation.
+  - **OpenEBS**: See [OpenEBS Add-on](/add-ons/openebs) in the kURL documentation and [What are the minimum requirements and supported container orchestrators?](https://openebs.io/docs/faqs/general#what-are-the-minimum-requirements-and-supported-container-orchestrators) in the OpenEBS documentation.
+
+- If you are migrating from Longhorn, complete the additional [Longhorn Prerequisites](#longhorn-prerequisites) below.
+
+### Longhorn Prerequisites
+
+If you are migrating from Longhorn to a different CSI provisioner, you must complete the following prerequisites in addition to the [General Prerequisites](#general-prerequisites) above:
+
+- Ensure that the version of Longhorn installed in your cluster is 1.2.0 or later or 1.3.0 or later. Longhorn versions 1.2.x and 1.3.x support Kubernetes versions 1.24 and earlier.
+
+- Confirm that the Longhorn volumes are in a healthy state. Run the following command to check the status of the volumes:
 
-_Or_, you can then migrate data _from_ Rook _to_ Minio and Longhorn using the following kURL spec:
+    ```
+    kubectl get volumes.longhorn.io -A
+    ```
+
+    If any volumes are reported as not healthy in the `Robustness` column in the ouput of this command, resolve the issue before proceeding.
+
+    To learn more about volume health, you can also inspect each volume individually:
+
+    ```
+    kubectl get volumes.longhorn.io -n longhorn-system <volume name> -o yaml
+    ```
+
+    In many cases, volume health is caused by issues with volume replication. Specifically, when multiple replicas are configured for a volume but not all have been scheduled. 
+
+    _**Note**_: During the data migration process in single-node clusters, the system automatically scales down the number of replicas to 1 in all Longhorn volumes to ensure the volumes are in a healthy state before beginning the data transfer. This is done to minimize the risk of a migration failure.
+
+- Confirm that Longhorn nodes are in a healthy state. The nodes must be healthy to ensure they are not over-provisioned and can handle scheduled workloads. Run the following command to check the status of the Longhorn nodes:
+
+    ```
+    kubectl get nodes.longhorn.io -A
+    ```
+
+    If any node is not reported as "Ready" and "Schedulable" in the output of this command, resolve the issue before proceeding.
+    
+    To learn more, you can also inspect each node individually and view its "Status" property:
+
+    ```
+    kubectl get nodes.longhorn.io -n longhorn-system <node name> -o yaml
+    ```
+
+- (OpenEBS Only) Before you migrate from Longhorn to OpenEBS:
```

Comment:
There were a few different places in the previous content that mentioned requirements related to being able to run both provisioners simultaneously. I wasn't totally certain if that applied specifically to migrating off Longhorn, or if it applies generically to anytime your are migrating from one provisioner to another.

That said, perhaps we don't need these `(OpenEBS Only)` and `(Rook Only)` prereqs if they are in fact actually already covered by the `Ensure that your cluster has adequate hardware resources to run both...` prereq in the General Prerequisites section above.

Reply:
That is true for all migrations, the provisioner must be supported in the Kubernetes version the user is migrating to. I have made a small adjustment to reflect that in the `General Prerequisites` section.